compose/smash/Dockerfile

```diff
@@ -1,19 +1,29 @@
-ARG UBUNTU_VERSION=20.04
+FROM nixos/nix AS nix-configured
+# prepare the base nix image with our nix.conf
+# this is done in a separate stage since we want nix.conf picked up in the next stage
 
-FROM ubuntu:${UBUNTU_VERSION}
+RUN nix-channel --update
+COPY ./compose/smash/nix/nix.conf /etc/nix/nix.conf
+
+FROM nix-configured as nix-builder
 
-ARG DBSYNC_BINARY_URL="https://update-cardano-mainnet.iohk.io/cardano-db-sync/cardano-db-sync-${CARDANO_DB_SYNC_VERSION:-13.1.0.0}-linux.tar.gz"
 ENV NETWORK=${NETWORK:-"mainnet"}
 
-RUN apt-get update && apt-get install curl -y
-RUN curl ${DBSYNC_BINARY_URL} --output /tmp/dbsync-binary.tar.gz &&\
-  tar --extract --file /tmp/dbsync-binary.tar.gz --directory /bin ./cardano-smash-server &&\
-  rm /tmp/dbsync-binary.tar.gz
+# checkout the source
+RUN git clone --depth 1 --branch ${CARDANO_DB_SYNC_VERSION:-13.1.0.0} \
```

Comment:
Instead of:

```
git clone https://github.com/some-org/some-repo \
  && cd some-repo \
  && nix build .#something
```

you can do just:

```
nix build github:some-org/some-repo#something
```

Reply:
@michalrus done in a908181e4f6b6d438a66cb2f8da5844425f3f8dc